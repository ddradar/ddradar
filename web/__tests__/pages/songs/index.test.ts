import { testSongList as songs } from '@ddradar/core/__tests__/data'
import Oruga from '@oruga-ui/oruga-next'
import { bulmaConfig } from '@oruga-ui/theme-bulma'
import { RouterLinkStub } from '@vue/test-utils'
import { describe, expect, test, vi } from 'vitest'
import { ref } from 'vue'

import { mountAsync } from '~/__tests__/test-utils'
import Page from '~/pages/songs/index.vue'

describe('Page /songs', () => {
  const query = { series: '10' }

  describe('snapshot tests', () => {
    test('{ isLoading: true } renders loading state', async () => {
      // Arrange
      /* eslint-disable @typescript-eslint/no-explicit-any */
      vi.mocked(useRoute).mockReturnValue({ query } as any)
      vi.mocked(useFetch).mockResolvedValue({
        pending: ref(true),
        data: ref([]),
      } as any)
      /* eslint-enable @typescript-eslint/no-explicit-any */

      // Act
      const wrapper = await mountAsync(Page, {
        global: {
          plugins: [[Oruga, bulmaConfig]],
          stubs: { NuxtLink: RouterLinkStub },
        },
      })

      // Assert
      expect(wrapper.element).toMatchSnapshot()
    })
    test('{ isLoading: false, songs: <Data> } renders song list', async () => {
      // Arrange
      /* eslint-disable @typescript-eslint/no-explicit-any */
      vi.mocked(useRoute).mockReturnValue({ query } as any)
      vi.mocked(useFetch).mockResolvedValue({
        pending: ref(false),
        data: ref(songs),
      } as any)
      /* eslint-enable @typescript-eslint/no-explicit-any */

      // Act
      const wrapper = await mountAsync(Page, {
        global: {
          plugins: [[Oruga, bulmaConfig]],
          stubs: { NuxtLink: RouterLinkStub },
        },
      })

      // Assert
      expect(wrapper.element).toMatchSnapshot()
    })
    test('{ isLoading: false, courses: [] } renders no data', async () => {
      // Arrange
      /* eslint-disable @typescript-eslint/no-explicit-any */
      vi.mocked(useRoute).mockReturnValue({ query } as any)
      vi.mocked(useFetch).mockResolvedValue({
        pending: ref(false),
        data: ref([]),
      } as any)
      /* eslint-enable @typescript-eslint/no-explicit-any */

      // Act
      const wrapper = await mountAsync(Page, {
        global: {
          plugins: [[Oruga, bulmaConfig]],
          stubs: { NuxtLink: RouterLinkStub },
        },
      })

      // Assert
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  test.each([
    [undefined, undefined, 'すべての楽曲を表示'],
    [undefined, '18', 'DanceDanceRevolution A3'],
    ['1', undefined, 'か'],
  ])('?name=%s&series=%s renders "%s" title', async (name, series, title) => {
    // Arrange
    /* eslint-disable @typescript-eslint/no-explicit-any */
    vi.mocked(useRoute).mockReturnValue({ query: { name, series } } as any)
    vi.mocked(useFetch).mockResolvedValue({
      pending: ref(true),
      data: ref([]),
    } as any)
    /* eslint-enable @typescript-eslint/no-explicit-any */

    // Act
    const wrapper = await mountAsync(Page, {
      global: {
        plugins: [[Oruga, bulmaConfig]],
        stubs: { NuxtLink: RouterLinkStub },
      },
    })

    // Assert
    expect(wrapper.find('h1').text()).toBe(title)
  })
})
