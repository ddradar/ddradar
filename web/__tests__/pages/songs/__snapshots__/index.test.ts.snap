// Vitest Snapshot v1

exports[`Page /songs > snapshot tests > { isLoading: false, courses: [] } renders no data 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    DDR X
  </h1>
  <div
    class="buttons"
  >
    
    <a
      class="button is-info"
    >
      DDR 1st
    </a>
    <a
      class="button is-info"
    >
      DDR 2ndMIX
    </a>
    <a
      class="button is-info"
    >
      DDR 3rdMIX
    </a>
    <a
      class="button is-info"
    >
      DDR 4thMIX
    </a>
    <a
      class="button is-info"
    >
      DDR 5thMIX
    </a>
    <a
      class="button is-info"
    >
      DDRMAX
    </a>
    <a
      class="button is-info"
    >
      DDRMAX2
    </a>
    <a
      class="button is-info"
    >
      DDR EXTREME
    </a>
    <a
      class="button is-info"
    >
      DDR SuperNOVA
    </a>
    <a
      class="button is-info"
    >
      DDR SuperNOVA2
    </a>
    <a
      class="button is-info is-outlined"
      disabled="true"
    >
      DDR X
    </a>
    <a
      class="button is-info"
    >
      DDR X2
    </a>
    <a
      class="button is-info"
    >
      DDR X3 VS 2ndMIX
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution (2013)
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution (2014)
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A20
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A20 PLUS
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A3
    </a>
    
  </div>
  <div
    class="b-table"
  >
    <div
      style="display: none;"
    >
      
      <span
        data-id="1"
      >
        Series
      </span>
      <span
        data-id="2"
      >
        Name
      </span>
      <span
        data-id="3"
      >
        Artist
      </span>
      <span
        data-id="4"
      >
        BPM
      </span>
      
    </div>
    <!--v-if-->
    <!--v-if-->
    <div
      class="table-wrapper"
    >
      <table
        class="table is-striped is-empty"
        tabindex="false"
      >
        <!--v-if-->
        <thead>
          <tr>
            <!--v-if-->
            <!--v-if-->
            
            <th
              class=""
              draggable="false"
            >
              <span>
                Series 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                Name 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                Artist 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                BPM 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            
            <!--v-if-->
          </tr>
          <!--v-if-->
          <!--v-if-->
        </thead>
        <tbody>
          
          
          <tr>
            <td
              colspan="4"
            >
              
              <section
                class="section"
              >
                <div
                  class="content has-text-grey has-text-centered"
                >
                  <p>
                    No Data
                  </p>
                </div>
              </section>
              
            </td>
          </tr>
        </tbody>
        <!--v-if-->
      </table>
      
      <transition-stub>
        <!--v-if-->
      </transition-stub>
      
    </div>
    
    <div
      class="level"
    >
      <div>
        
        
      </div>
      <div>
        <div>
          <nav
            class="pagination"
          >
            <a
              aria-current="false"
              class="pagination-link pagination-previous is-disabled pagination-previous is-disabled"
              href="#"
              role="button"
            >
              
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="mdi mdi-chevron-left mdi-24px"
                />
              </span>
              
            </a>
            <a
              aria-current="false"
              class="pagination-link pagination-next is-disabled pagination-next is-disabled"
              href="#"
              role="button"
            >
              
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="mdi mdi-chevron-right mdi-24px"
                />
              </span>
              
            </a>
            <ul
              class="pagination-list"
            >
              <!--First-->
              <!--v-if-->
              <!--v-if-->
              <!--Pages-->
              
              
              <!--Last-->
              <!--v-if-->
              <!--v-if-->
            </ul>
          </nav>
        </div>
      </div>
    </div>
    
  </div>
</section>
`;

exports[`Page /songs > snapshot tests > { isLoading: false, songs: <Data> } renders song list 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    DDR X
  </h1>
  <div
    class="buttons"
  >
    
    <a
      class="button is-info"
    >
      DDR 1st
    </a>
    <a
      class="button is-info"
    >
      DDR 2ndMIX
    </a>
    <a
      class="button is-info"
    >
      DDR 3rdMIX
    </a>
    <a
      class="button is-info"
    >
      DDR 4thMIX
    </a>
    <a
      class="button is-info"
    >
      DDR 5thMIX
    </a>
    <a
      class="button is-info"
    >
      DDRMAX
    </a>
    <a
      class="button is-info"
    >
      DDRMAX2
    </a>
    <a
      class="button is-info"
    >
      DDR EXTREME
    </a>
    <a
      class="button is-info"
    >
      DDR SuperNOVA
    </a>
    <a
      class="button is-info"
    >
      DDR SuperNOVA2
    </a>
    <a
      class="button is-info is-outlined"
      disabled="true"
    >
      DDR X
    </a>
    <a
      class="button is-info"
    >
      DDR X2
    </a>
    <a
      class="button is-info"
    >
      DDR X3 VS 2ndMIX
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution (2013)
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution (2014)
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A20
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A20 PLUS
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A3
    </a>
    
  </div>
  <div
    class="b-table"
  >
    <div
      style="display: none;"
    >
      
      <span
        data-id="1"
      >
        Series
      </span>
      <span
        data-id="2"
      >
        Name
      </span>
      <span
        data-id="3"
      >
        Artist
      </span>
      <span
        data-id="4"
      >
        BPM
      </span>
      
    </div>
    <!--v-if-->
    <!--v-if-->
    <div
      class="table-wrapper"
    >
      <table
        class="table is-striped"
        tabindex="false"
      >
        <!--v-if-->
        <thead>
          <tr>
            <!--v-if-->
            <!--v-if-->
            
            <th
              class=""
              draggable="false"
            >
              <span>
                Series 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                Name 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                Artist 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                BPM 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            
            <!--v-if-->
          </tr>
          <!--v-if-->
          <!--v-if-->
        </thead>
        <tbody>
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="Series"
              scoped=""
            >
              1st
            </td>
            <td
              class=""
              data-label="Name"
              scoped=""
            >
              <a>
                PARANOiA
              </a>
            </td>
            <td
              class=""
              data-label="Artist"
              scoped=""
            >
              180
            </td>
            <td
              class=""
              data-label="BPM"
              scoped=""
            >
              180
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="Series"
              scoped=""
            >
              X
            </td>
            <td
              class=""
              data-label="Name"
              scoped=""
            >
              <a>
                PARANOiA(X-Special)
              </a>
            </td>
            <td
              class=""
              data-label="Artist"
              scoped=""
            >
              180
            </td>
            <td
              class=""
              data-label="BPM"
              scoped=""
            >
              180
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="Series"
              scoped=""
            >
              X
            </td>
            <td
              class=""
              data-label="Name"
              scoped=""
            >
              <a>
                SP-TRIP MACHINE～JUNGLE MIX～(X-Special)
              </a>
            </td>
            <td
              class=""
              data-label="Artist"
              scoped=""
            >
              DE-SIRE
            </td>
            <td
              class=""
              data-label="BPM"
              scoped=""
            >
              160
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <!--v-if-->
        </tbody>
        <!--v-if-->
      </table>
      
      <transition-stub>
        <!--v-if-->
      </transition-stub>
      
    </div>
    
    <div
      class="level"
    >
      <div>
        
        
      </div>
      <div>
        <div>
          <nav
            class="pagination"
          >
            <a
              aria-current="false"
              class="pagination-link pagination-previous is-disabled pagination-previous is-disabled"
              href="#"
              role="button"
            >
              
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="mdi mdi-chevron-left mdi-24px"
                />
              </span>
              
            </a>
            <a
              aria-current="false"
              class="pagination-link pagination-next is-disabled pagination-next is-disabled"
              href="#"
              role="button"
            >
              
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="mdi mdi-chevron-right mdi-24px"
                />
              </span>
              
            </a>
            <ul
              class="pagination-list"
            >
              <!--First-->
              <!--v-if-->
              <!--v-if-->
              <!--Pages-->
              
              <li>
                <a
                  aria-current="true"
                  class="pagination-link is-current"
                  href="#"
                  role="button"
                >
                  
                  1
                  
                </a>
              </li>
              
              <!--Last-->
              <!--v-if-->
              <!--v-if-->
            </ul>
          </nav>
        </div>
      </div>
    </div>
    
  </div>
</section>
`;

exports[`Page /songs > snapshot tests > { isLoading: true } renders loading state 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    DDR X
  </h1>
  <div
    class="buttons"
  >
    
    <a
      class="button is-info"
    >
      DDR 1st
    </a>
    <a
      class="button is-info"
    >
      DDR 2ndMIX
    </a>
    <a
      class="button is-info"
    >
      DDR 3rdMIX
    </a>
    <a
      class="button is-info"
    >
      DDR 4thMIX
    </a>
    <a
      class="button is-info"
    >
      DDR 5thMIX
    </a>
    <a
      class="button is-info"
    >
      DDRMAX
    </a>
    <a
      class="button is-info"
    >
      DDRMAX2
    </a>
    <a
      class="button is-info"
    >
      DDR EXTREME
    </a>
    <a
      class="button is-info"
    >
      DDR SuperNOVA
    </a>
    <a
      class="button is-info"
    >
      DDR SuperNOVA2
    </a>
    <a
      class="button is-info is-outlined"
      disabled="true"
    >
      DDR X
    </a>
    <a
      class="button is-info"
    >
      DDR X2
    </a>
    <a
      class="button is-info"
    >
      DDR X3 VS 2ndMIX
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution (2013)
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution (2014)
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A20
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A20 PLUS
    </a>
    <a
      class="button is-info"
    >
      DanceDanceRevolution A3
    </a>
    
  </div>
  <div
    class="b-table"
  >
    <div
      style="display: none;"
    >
      
      <span
        data-id="1"
      >
        Series
      </span>
      <span
        data-id="2"
      >
        Name
      </span>
      <span
        data-id="3"
      >
        Artist
      </span>
      <span
        data-id="4"
      >
        BPM
      </span>
      
    </div>
    <!--v-if-->
    <!--v-if-->
    <div
      class="table-wrapper"
    >
      <table
        class="table is-striped is-empty"
        tabindex="false"
      >
        <!--v-if-->
        <thead>
          <tr>
            <!--v-if-->
            <!--v-if-->
            
            <th
              class=""
              draggable="false"
            >
              <span>
                Series 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                Name 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                Artist 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                BPM 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            
            <!--v-if-->
          </tr>
          <!--v-if-->
          <!--v-if-->
        </thead>
        <tbody>
          
          
          <tr>
            <td
              colspan="4"
            >
              
              <section
                class="section"
              >
                <div
                  class="b-skeleton is-animated"
                >
                  <div
                    class="b-skeleton-item is-rounded"
                  />
                  <div
                    class="b-skeleton-item is-rounded"
                  />
                  <div
                    class="b-skeleton-item is-rounded"
                  />
                </div>
              </section>
              
            </td>
          </tr>
        </tbody>
        <!--v-if-->
      </table>
      
      <transition-stub>
        <div
          class="o-load loading"
        >
          <div
            class="o-load__overlay loading-overlay"
          />
          
          <span
            class="icon is-spin is-medium o-load__icon icon"
          >
            <i
              class="mdi mdi-loading mdi-36px"
            />
          </span>
          
        </div>
      </transition-stub>
      
    </div>
    
    <div
      class="level"
    >
      <div>
        
        
      </div>
      <div>
        <div>
          <nav
            class="pagination"
          >
            <a
              aria-current="false"
              class="pagination-link pagination-previous is-disabled pagination-previous is-disabled"
              href="#"
              role="button"
            >
              
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="mdi mdi-chevron-left mdi-24px"
                />
              </span>
              
            </a>
            <a
              aria-current="false"
              class="pagination-link pagination-next is-disabled pagination-next is-disabled"
              href="#"
              role="button"
            >
              
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="mdi mdi-chevron-right mdi-24px"
                />
              </span>
              
            </a>
            <ul
              class="pagination-list"
            >
              <!--First-->
              <!--v-if-->
              <!--v-if-->
              <!--Pages-->
              
              
              <!--Last-->
              <!--v-if-->
              <!--v-if-->
            </ul>
          </nav>
        </div>
      </div>
    </div>
    
  </div>
</section>
`;
